centinel/client.py

```diff
@@ -123,8 +125,46 @@ def run(self, data_dir=None):
 
             try:
                 logging.info("Running %s test." % (name))
+
+                root = True
+                if os.geteuid() != 0:
+                    logging.info("Centinel is not running as root, "
+                                 "tcpdump will not start.")
+                    root = False
+
+                td = Tcpdump()
+                tcpdump_started = False
+                try:
+                    if root:
+                        td.start()
+                        tcpdump_started = True
+                        logging.info("tcpdump started...")
+                except Exception as e:
+                    logging.warning("Failed to run tcpdump: %s" %(e))
+
                 exp = Exp(input_file)
                 exp.run()
+
+                if tcpdump_started:
+                    logging.info("Waiting for tcpdump to process packets...")
+                    # 5 seconds should be enough. this hasn't been tested on
+                    # a RaspberryPi or a Hummingboard i2
+                    time.sleep(5)
```

Comment:
@ben-jones, before you ask, this is because if you terminate `tcpdump` prematurely, it will not have enough time to process packets and you will end up with an empty pcap. I do not have a way of knowing when it has finished processing, so I'll just wait 5 seconds.